src/main/kotlin/io/github/mojira/arisa/modules/RemoveVersionModule.kt

```diff
@@ -0,0 +1,39 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import io.github.mojira.arisa.domain.Issue
+import java.time.Instant
+
+class RemoveVersionModule(
+) : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val addedVersions = getExtraVersionsLatelyAddedByNonVolunteers(lastRun)
+            val removeAddedVersions = affectedVersions
+                .filter { it.id in addedVersions }
+                .map { it.remove }
+            assertNotEmpty(removeAddedVersions).bind()
+            removeAddedVersions.forEach(::run)
+        }
+    }
+
+    private fun Issue.getExtraVersionsLatelyAddedByNonVolunteers(lastRun: Instant): List<String> =
+        if (created.isAfter(lastRun) && (resolution == "Invalid" || resolution == "Won't Fix" || resolution == "Works As Intended" || resolution == "Duplicate" || resolution == "Incomplete")) {
+            if (isVolunteer(reporter?.getGroups?.invoke())) {
+                emptyList()
+            } else {
+                affectedVersions.map { ver -> ver.id }
+            }
+        } else {
+            changeLog
+                .asSequence()
+                .filter { it.created.isAfter(lastRun) }
+                .filter { it.field.toLowerCase() == "version" }

```

Comment:
It appears even in Kotlin `toLowerCase()` uses the default locale so it would be good to specify `Locale.ENGLISH` / `Locale.ROOT`

Reply:
That's how the FutureVersionModule did it, and I copied it from there

Reply:
Added it anyways

Reply:
It seems that I can't do it. And since it's not completely necessary due to other modules using these functions without using the locale, I'm just not going to do it

Reply:
Hmm, yes it appears the project uses everywhere else `toLowerCase()` without locale as well. At least in Java you have to (sadly) always specify the locale since the default locale could have special lower casing rules (e.g. Turkish). Though it is unlikely that Arisa runs on a system with such a default locale, it is good to be on the safe side nonetheless.
Maybe @urielsalis (who seems to have much experience with Kotlin) knows if specifying a locale is necessary for Kotlin as well? Though it appears to be the case.

Reply:
Generally yes, but we control the server(and we are about to put it on docker) and it's always set to UTC and English

Somewhere in the future we should probably add it everywhere